coreneuron/sim/fadvance_core.cpp

```diff
@@ -30,6 +30,45 @@ extern corenrn_parameters corenrn_param;
 static void* nrn_fixed_step_thread(NrnThread*);
 static void* nrn_fixed_step_group_thread(NrnThread*, int, int, int&);
 
+
+namespace {
+
+class ProgressBar {
+    bool show;
+    progressbar* pbar;
+    int current_step = 0;
+    constexpr static int progressbar_update_steps = 5;
+
+  public:
+    ProgressBar(int nsteps)
+        : show(nrnmpi_myid == 0 && !corenrn_param.is_quiet()) {
+        if (show) {
+            printf("\n");
+            pbar = progressbar_new("psolve", nsteps);
+        }
+    }
+
+    void update(int step, double time) {
+        current_step = step;
+        if (show && (current_step % progressbar_update_steps) == 0) {
+            progressbar_update(pbar, current_step, time);
+        }
+    }
+
+    void step(double time) {
+        update(current_step + 1, time);
+    }
+
+    virtual ~ProgressBar() {
```

Comment:
Nitpick: why `virtual`?

Reply:
Just because it's a good habit. But in this case maybe it's not useful.